File: Java/Solenoid Single/src/main/java/frc/robot/Robot.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package frc.robot;

import edu.wpi.first.wpilibj.PneumaticsModuleType;
import edu.wpi.first.wpilibj.Solenoid;
import edu.wpi.first.wpilibj.TimedRobot;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;


public class Robot extends TimedRobot {
  Solenoid m_solenoid = new Solenoid(PneumaticsModuleType.REVPH, 0);
  @Override
  public void robotInit() {
    // Add Set Solenoid button to Shuffleboard
   SmartDashboard.setDefaultBoolean("Set Solenoid", false);
  }

  @Override
  public void robotPeriodic() {
    /**
     * Get the state of the solenoid.
     */
    SmartDashboard.putBoolean("Get Solenoid", m_solenoid.get());
  }

  @Override
  public void autonomousInit() {}

  @Override
  public void autonomousPeriodic() {}

  @Override
  public void teleopInit() {}

  @Override
  public void teleopPeriodic() {
    /**
     * Set the solenoid output.
     */
    m_solenoid.set(SmartDashboard.getBoolean("Set Solenoid", false));
  }

  @Override
  public void disabledInit() {}

  @Override
  public void disabledPeriodic() {}

  @Override
  public void testInit() {}

  @Override
  public void testPeriodic() {}
}

File: C++/Solenoid Single/src/main/cpp/Robot.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "Robot.h"

#include <frc/smartdashboard/SmartDashboard.h>

void Robot::RobotInit() {
  // Add Set Solenoid button to Shuffleboard
  frc::SmartDashboard::SetDefaultBoolean("Set Solenoid", false);
}

void Robot::RobotPeriodic() {
  /**
   * Get the state of the solenoid.
   */
  frc::SmartDashboard::PutBoolean("Get Solenoid", m_solenoid.Get());
}

void Robot::AutonomousInit() {}

void Robot::AutonomousPeriodic() {}

void Robot::TeleopInit() {}

void Robot::TeleopPeriodic() {
  /**
   * Set the solenoid output.
   */
  m_solenoid.Set(frc::SmartDashboard::GetBoolean("Set Solenoid", false));
}

void Robot::DisabledInit() {}

void Robot::DisabledPeriodic() {}

void Robot::TestInit() {}

void Robot::TestPeriodic() {}

#ifndef RUNNING_FRC_TESTS
int main() { return frc::StartRobot<Robot>(); }
#endif
